pkg/envoy/cell.go

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package envoy
+
+import (
+	"fmt"
+
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/ipcache"
+	"github.com/cilium/cilium/pkg/option"
+)
+
+// Cell initializes and manages the Envoy proxy and its control-plane components like xDS- and accesslog server.
+// It is used to provide support for Ingress, GatewayAPI and L7 network policies (e.g. HTTP).
+var Cell = cell.Module(
+	"envoy-proxy",
+	"Envoy proxy and control-plane",
+
+	cell.Provide(newEnvoyXDSServer),
+	cell.Invoke(registerEnvoyAccessLogServer),
+)
+
+type xdsServerParams struct {
+	cell.In
+
+	Lifecycle hive.Lifecycle
+	IPCache   *ipcache.IPCache
+}
+
+func newEnvoyXDSServer(params xdsServerParams) (*XDSServer, error) {
+	xdsServer, err := newXDSServer(GetSocketDir(option.Config.RunDir), params.IPCache)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Envoy xDS server: %w", err)
+	}
+
+	if !option.Config.EnableL7Proxy {
+		log.Debug("L7 proxies are disabled - don't start Envoy xDS server")
+		return xdsServer, nil
```

Comment:
What's the behavior of the public API methods of `XDSServer` when it hasn't been started?

Also could we hide `EnableL7Proxy` option inside this package (or pkg/proxy) and refactor the users of this API to not care whether or not it's been enabled? E.g. by either always calling a no-op method or adding `IsEnabled` or similar if there's expensive work needed prior to a call. I think it'd be preferably to reduce reliance on checking "option.Config.EnableFoo" and instead make APIs that are safe to use regardless of whether they're enabled or not.

Reply:
> What's the behavior of the public API methods of XDSServer when it hasn't been started?

it's just updating it's internal state without actually exposing the config via gRPC.

> Also could we hide EnableL7Proxy option inside this package (or pkg/proxy) and refactor the users of this API to not care whether or not it's been enabled? E.g. by either always calling a no-op method or adding IsEnabled or similar if there's expensive work needed prior to a call. I think it'd be preferably to reduce reliance on checking "option.Config.EnableFoo" and instead make APIs that are safe to use regardless of whether they're enabled or not.

I'd like to refactor this in a follow up PR too. I already pointed this out here https://github.com/cilium/cilium/pull/26627#discussion_r1252180307. But this needs to be handled in the proxy cell - which should return a noop-proxy in case of L7 proxy functionality is disabled.